packages/ui/src/theme/utils.ts

```diff
@@ -48,3 +51,95 @@ interface NameTransformProps {
 export function cssNameTransform({ path = [] }: NameTransformProps): string {
   return `${kebabCase([CSS_VARIABLE_PREFIX, ...path].join(' '))}`;
 }
+
+export type pathObject = {
+  path: string[];
+};
+
+export const STATE_NAMES = [
+  'hover',
+  'focus',
+  'active',
+  'loading',
+  'disabled',
+  'visited',
+];
+
+export function findDeprecatedStateTokens(baseTheme: BaseTheme): pathObject[] {
+  const baseEntry = baseTheme?.tokens || {};
+  return traverse(['', baseEntry], []);
```

Comment:
question: what's the first element, empty string for? 

Can we do this?

```suggestion
  return traverse([baseEntry], []);
```